content/en/guide/v10/typescript.md

```diff
@@ -194,52 +251,59 @@ export class Button extends Component {
 }
 ```
 
-You can restrict event handlers by adding a type annotation for `this` to the function signature as the first argument. This argument will be erased after transpilation.
+You can provide a more specific type than `EventTarget` by using [this parameters](https://www.typescriptlang.org/docs/handbook/functions.html#this-parameters):
 
 ```tsx
+import { h, Component } from 'preact';
+
 export class Button extends Component {
-  // Adding the this argument restricts binding
+  // Define `this parameters`
   handleClick(this: HTMLButtonElement, event: MouseEvent) {
-    event.preventDefault();
-    if (event.target instanceof HTMLElement) {
-      console.log(event.target.localName); // "button"
-    }
+    console.log(this.tagName); // "BUTTON"
   }
 
   render() {
-    return (
-      <button onClick={this.handleClick}>{this.props.children}</button>
-    );
+    return <button onClick={this.handleClick}>{this.props.children}</button>;
+  }
+}
+```
+
+You can provide a more specific type than `EventTarget` by using `TargetedEvent`. `TargetedEvent` type has two generic type parameters which corresponds `currentTarget`'s type and Event type:
+
+```tsx
+import { h, Component, JSX } from 'preact';
+
+export class Button extends Component {
+  handleClick({ currentTarget }: JSX.TargetedEvent<HTMLInputElement, Event> {
+    console.log(currentTarget.tagName); // "BUTTON"
+  }
+
+  render() {
+    return <button onClick={this.handleClick}>{this.props.children}</button>;
   }
 }
 ```
 
-## Typing references
+## References
 
-The `createRef` function is also generic, and lets you bind references to element types. In this example, we ensure that the reference can only be bound to `HTMLAnchorElement`. Using `ref` with any other element lets TypeScript thrown an error:
```

Comment:
>  Using `ref` with any other element lets TypeScript thrown an error.

This is explanation about TypeScript `TypeError`.
I think it is unnecessary in Preact document.
Therefore, I removed this sentence.